### ISSUE TITLE
Use of angle brackets around file names for include statements

### ISSUE THREAD

**Issue by elfring** [eclipse/omr#13]
Would you like to [replace any double quotes](https://github.com/eclipse/omr/blob/b3c44edf20ca4ea9362b9325754d1eed85a34394/include_core/OMR_VM.hpp#L22 "Check header file inclusion style.") by [angle brackets around file names for include statements](http://stackoverflow.com/questions/21593/what-is-the-difference-between-include-filename-and-include-filename "What is the difference between #include <filename> and #include “filename”?")?

**Comment by charliegracie** [comment]
Currently the Eclipse OMR project uses <> for system include files and "" for our project header files. I believe we are not violating any spec with this usage pattern but if there is a best practice we could be following that is something we would consider. Do you have any more details?

**Comment by elfring** [comment]
I suggest to reconsider the consequences of [the following wording](http://c0x.coding-guidelines.com/6.10.2.html "Source file inclusion") from [the section "16.2 Source file inclusion"](http://open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3797.pdf "C++ working draft") in the standard specification for the programming language "C++".
```
…
The named source file is searched for in an implementation-defined manner. If this search is not supported, or if the search fails, the directive is reprocessed as if it read

#include <h-char-sequence> new-line
…
```

* How do you think about to avoid "a duplicated file search" here?
* Would you like to [restrict the searched directories](https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html "Include syntax description") for header files of your software?

**Comment by mstoodle** [comment]
Boiling that down: the point would be to avoid searching the "current directory" for our headers by using angle brackets rather than the quoted form? Given the fuzziness of the notion of the "current" directory when compiling, I can sort of see the argument. But I also think most people would find the change odd.

Anyone feel strongly we should go down this path? Otherwise I vote we keep it as is at least for now; we've managed to go 20+ years without running into problems with the current directory on the search path...

**Comment by elfring** [comment]
>Given the fuzziness of the notion of the "current" directory when compiling, I can sort of see the argument.

I would generally prefer to exclude the potential for the inclusion of header files from unexpected directories. Can such a security detail become a bit more important occasionally?

**Comment by mstoodle** [comment]
Angela pointed out that the same standard specification for C++ programming you pulled text from also contains the following recommendation (point 7 on page 403):

[ Note: Although an implementation may provide a mechanism for making arbitrary source files available to the < > search, in general programmers should use the < > form for headers provided with the implementa- tion, and the " " form for sources outside the control of the implementation. For instance:
     #include <stdio.h>
     #include <unistd.h>
     #include "usefullib.h"
     #include "myprog.h"
— end note ]

OMR makefiles assume they're being run in the current directory and would break for a number of reasons if they weren't being run there.

The security argument feels pretty weak to me: any security risk would come from someone with access to the platform to rebuild OMR / language runtime and they could just as easily modify the makefile to include whatever directory they want on the include path. Security details are definitely important, but I don't see this recommendation providing any real security improvement.

In summary, using angle brackets on OMR header includes would seem to serve no purpose other than to specifically go against the general recommendation Angela pointed out above, which virtually every C/C++ programmer is used to.

I think we should keep things as they are.
